### PR TITLE
restore query arguments if wagtail's redirect process strips them

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/7072
Also See https://github.com/wagtail/wagtail/issues/7339

test urls:
- https://foundation-s-fix-redire-emlcd9.herokuapp.com/regrets?utm_campaign=youtuberegrets&utm_source=twitter&utm_medium=social&utm_content=1625741125
- https://foundation-s-fix-redire-emlcd9.herokuapp.com/campaigns/single-page/?subscribed=1&utm_source=email&utm_medium=email&utm_campaign=2020advocacy-fr&utm_content=appleidfathanks&utm_term=5382510

### CODE REVIEW

File: network-api/networkapi/wagtailcustomization/redirects/middleware.py [hunk 1]
```diff
@@ -11,10 +13,24 @@ def _new_get_redirect(request, path):
...
-    return _original_get_redirect(request, path)
+    redirect = _original_get_redirect(request, path)
+
+    # Wagtail currently does not forward query arguments, so for
+    # any URL with stripped query arguments we make a new, on-the-fly
+    # redirect with the same path/site bindings, but an updated link
+    # with the query arguments restored.
+    if redirect and request.GET:
+        target = redirect.link
+        if request.GET and "?" not in target:
```
Comment (Pomax) [inline network-api/networkapi/wagtailcustomization/redirects/middleware.py:26]:
```suggestion
        if "?" not in target:
```

File: network-api/networkapi/wagtailcustomization/redirects/middleware.py [hunk 2]
```diff
@@ -11,10 +13,24 @@ def _new_get_redirect(request, path):
...
+    # redirect with the same path/site bindings, but an updated link
+    # with the query arguments restored.
+    if redirect and request.GET:
+        target = redirect.link
+        if "?" not in target:
+            redirect = Redirect(
+                old_path=redirect.old_path,
+                site=redirect.site,
+                redirect_link=f'{redirect.link}?{urlencode(request.GET)}'
+            )
```
Comment (Pomax) [inline network-api/networkapi/wagtailcustomization/redirects/middleware.py:31]:
```suggestion
    if redirect and request.GET and "?" not in redirect.link:
        redirect = Redirect(
            old_path=redirect.old_path,
            site=redirect.site,
            redirect_link=f'{redirect.link}?{urlencode(request.GET)}'
        )
```

File: network-api/networkapi/wagtailcustomization/redirects/middleware.py [hunk 3]
```diff
@@ -11,10 +13,22 @@ def _new_get_redirect(request, path):
...
-        path = path + f'/?{request.META["QUERY_STRING"]}'
+
     # Then hand off processing to the original redirect logic.
-    return _original_get_redirect(request, path)
+    redirect = _original_get_redirect(request, path)
+
+    # Wagtail currently does not forward query arguments, so for
+    # any URL with stripped query arguments we make a new, on-the-fly
+    # redirect with the same path/site bindings, but an updated link
+    # with the query arguments restored.
```
Comment (Pomax) [inline network-api/networkapi/wagtailcustomization/redirects/middleware.py:23]:
```suggestion
    # Wagtail currently does not forward query arguments, so for
    # any URL with stripped query arguments we make a new, on-the-fly
    # redirect with the same path/site bindings, but an updated link
    # with the query arguments restored.
    #
    # See https://github.com/wagtail/wagtail/issues/7339 for more details.
```

**Review by TheoChevalier — APPROVED** [review]

**Review by danielfmiranda — APPROVED** [review]
Hi Pomax! Thanks for tagging me in this PR! I am appreciative of the comments that you have added which link to the wagtail issue regarding the redirects, it made it easy to follow and see what is being changed. 

Everything looks good to me, Approving!

**Pull request merged**